framework/src/actions/MeshOnlyAction.C

```diff
@@ -67,14 +69,92 @@ MeshOnlyAction::act()
   {
     TIME_SECTION("act", 1, "Writing Exodus");
 
-    ExodusII_IO exio(mesh_ptr->getMesh());
+    auto & output_mesh = mesh_ptr->getMesh();
+    ExodusII_IO exio(output_mesh);
 
     Exodus::setOutputDimensionInExodusWriter(exio, *mesh_ptr);
 
     // Default to non-HDF5 output for wider compatibility
     exio.set_hdf5_writing(false);
 
     exio.write(mesh_file);
+
+    // Check if extra element integers should be outputted to Exodus file
+    unsigned int n_eeid = output_mesh.n_elem_integers();
+    if (n_eeid > 0)
+    {
+      // Iterate through all actions and see if `Outputs/output_extra_element_ids = true` in input
+      // file
+      const auto & output_actions = _app.actionWarehouse().getActionListByName("add_output");
+      bool output_extra_ids = true;
```

Comment:
seems odd to default to true here and not in the AdvancedOutput parameter

Reply:
So this was an intentional design decision. If we set `output_extra_element_ids` to default to true in AdvancedOutput, then there are a large number of test cases that will fail. This occurs when the mesh has extra element id's that are defined on it that do not get added manually in the input file. If the parameter gets set to true by default then all of those extra element id's end up in the final exodus file and result in an Exodiff failure. I chose false to be the default beahvior in order to minimize the number of test failures that would occur (not only in MOOSE framework but also in all downstream MOOSE apps where such behavior is prevalent). Let me know if this is okay to keep as is or if you'd like me change the default AdvancedOutput parameter to true